Lib/test/datetimetester.py

```diff
@@ -656,11 +656,31 @@ def test_str(self):
     def test_repr(self):
         name = 'datetime.' + self.theclass.__name__
         self.assertEqual(repr(self.theclass(1)),
-                         "%s(1)" % name)
+                         "%s(days=1)" % name)
         self.assertEqual(repr(self.theclass(10, 2)),
-                         "%s(10, 2)" % name)
+                         "%s(days=10, seconds=2)" % name)
         self.assertEqual(repr(self.theclass(-10, 2, 400000)),
-                         "%s(-10, 2, 400000)" % name)
+                         "%s(days=-10, seconds=2, microseconds=400000)" % name)
+        self.assertEqual(repr(self.theclass(1, -1)),
+                         "%s(seconds=86399)" % name)
+        self.assertEqual(repr(self.theclass(0, 1, -1)),
+                         "%s(microseconds=999999)" % name)
+        self.assertEqual(repr(self.theclass(0, 86400)),
+                         "%s(days=1)" % name)
+        self.assertEqual(repr(self.theclass(0, 2 * 86400)),
+                         "%s(days=2)" % name)
+        self.assertEqual(repr(self.theclass(days=1, seconds=0)),
+                         "%s(days=1)" % name)
+        self.assertEqual(repr(self.theclass(seconds=60)),
+                         "%s(seconds=60)" % name)
+        self.assertEqual(repr(self.theclass()),
+                         "%s(seconds=0)" % name)
+        self.assertEqual(repr(self.theclass(microseconds=100)),
+                         "%s(microseconds=100)" % name)
+        self.assertEqual(repr(self.theclass(days=1, microseconds=100)),
+                         "%s(days=1, microseconds=100)" % name)
+        self.assertEqual(repr(self.theclass(seconds=1, microseconds=100)),
+                         "%s(seconds=1, microseconds=100)" % name)
```

Comment:
Some of the tests were added to differentiate the current implementation from the old one: i.e. no leading `days=0` or `seconds=0` and to highlight the current handling of negative values (i.e. the normalization of the attributes, as described in the docs). I'll remove the ones which don't test anything unique not covered elsewhere.